src/Functions/UserDefinedFunction.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <Functions/FunctionFactory.h>
+#include <Functions/IFunctionImpl.h>
+#include <Interpreters/Context.h>
+#include <Parsers/ASTCreateFunctionQuery.h>
+
+namespace DB
+{
+
+class UserDefinedFunction;
+using UserDefinedFunctionPtr = std::shared_ptr<UserDefinedFunction>;
+
+class UserDefinedFunction : public IFunction
```

Comment:
It may be logical to use `IFunctionOverloadResolver` interface.

This UDF is polymorphic, and can be used with different types. However, we don't need to resolve types for every processing block. `OverloadResolver` can check types and build ready `ExpressionActions`. Those actions should be used in `IFunctionBase`.

Reply:
Also, looks like we need to disable all `useDefaultImplementation` flags.

Reply:
We have specific values of arguments inside ExpressionActions. Why it will work if we build ExpressionAction once?